lib/ansible/modules/cloud/amazon/aws_elasticbeanstalk_app.py

```diff
@@ -0,0 +1,261 @@
+#!/usr/bin/python
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'
+                    }
+
+DOCUMENTATION = '''
+---
+module: aws_elasticbeanstalk_app
+
+short_description: create, update, delete and list beanstalk application
+
+version_added: "2.5"
+
+description:
+    - "creates, updates, deletes beanstalk applications if app_name is provided. Can also list applications"
+
+options:
+  app_name:
+    description:
+      - name of the beanstalk application you wish to manage
+    required: false
+    default: null
+    aliases: [ 'name' ]
```

Comment:
I would just call this name and not bother with the alias

Reply:
I've only included the _name_ alias to be consistent with other Ansible modules, e.g. apt having package, pkg as an alias to name. *app_name* is the term used by the api. Should we stay more consistent with the api? or more consistent with Ansible modules? I chose alias. Happy to follow your lead.

Reply:
I'll let the first bullet point of http://docs.ansible.com/ansible/latest/dev_guide/developing_modules_best_practices.html have the final say.

It's fine for it to accept an alias, so I withdraw my complaint :)